src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,176 @@
+<?php
+declare(strict_types=1);
+
+namespace Triniti\Sys;
+
+use Defuse\Crypto\Crypto;
+use Gdbots\Ncr\Exception\NodeNotFound;
+use Gdbots\Ncr\Ncr;
+use Gdbots\Pbj\Message;
+use Gdbots\Pbjx\CommandHandler;
+use Gdbots\Pbjx\Pbjx;
+use Gdbots\Schemas\Pbjx\StreamId;
+use Gdbots\UriTemplate\UriTemplateService;
+use Google\Service\SearchConsole\InspectUrlIndexResponse;
+use Psr\Log\LoggerInterface;
+use Triniti\Schemas\Sys\Event\SeoInspectedV1;
+use Defuse\Crypto\Key;
+
+
+final class InspectSeoHandler implements CommandHandler
+{
+    public static function handlesCuries(): array
+    {
+        return [
+            'triniti:sys:command:inspect-seo',
+        ];
+    }
+
+    public function __construct(
+        private readonly Ncr        $ncr,
+        private readonly Key        $key,
+        private readonly Flags      $flags,
+        private readonly LoggerInterface $logger,
+    )
+    {
+    }
+
+    public function handleCommand(Message $command, Pbjx $pbjx): void
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_disabled')) {
+            return;
+        }
+
+        $searchEngines = $command->get('search_engines', ['google']);
+        $nodeRef = $command->get('node_ref');
+
+        try {
+            $node = $this->ncr->getNode($nodeRef);
+        } catch (NodeNotFound $e) {
+            $this->logger->error('Unable to get node for search engines processing.', [
+                'exception' => $e,
+                'node_ref' => $nodeRef
+            ]);
+            return;
+        }
+
+        $retryCommand = clone $command;
+        $retryCommand->clear('search_engines');
+
+        foreach ($searchEngines as $searchEngine) {
+            $methodName = 'checkIndexStatusFor' . ucfirst($searchEngine);
+
+            if (!method_exists($this, $methodName)) {
+                continue;
+            }
+
+            $retryCommand = $this->$methodName($retryCommand, $pbjx, $node);
+        }
+
+        if (empty($retryCommand->get('search_engines'))) {
+            return;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+
+        if ($retries < $maxRetries) {
+            $retryCommand->set('ctx_retries', $retries + 1);
+
+            $pbjx->sendAt(
+                $retryCommand,
+                strtotime($this->flags->getString('inspect_seo_retry_delay', '+15 minutes')),
+                "$nodeRef.inspect-seo"
+            );
+        }
+    }
+
+    public function checkIndexStatusForGoogle(Message $command, Pbjx $pbjx, Message $node): Message
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_google_disabled')) {
+            return $command;
+        }
+
+        $request = new \Google_Service_SearchConsole_InspectUrlIndexRequest();
+        $siteUrl = $this->flags->getString('inspect_seo_google_site_url');
```

Comment:
@efjacobson @digitalartcrew don't we know site url from the resolved template url (canonical) for a given article?  shouldn't need another flag.

Reply:
@gdbrown siteUrl references the value ` sc-domain:example.com ` generated by Google Search Console not the url to be inspected. We could do something `sc-domain:{siteUrl}` if maybe that is what you're suggesting versus a flag

Reply:
@digitalartcrew i figured it was,  but the resolved canonical, you can get site url from that value.  i'd just use parse_url() and get the host from the canonical.

Reply:
@gdbrown fine by me but do note that this would now make an assumption about what kind of property (domain vs url-prefix) it is in search console

ref https://developers.google.com/webmaster-tools/v1/urlInspection.index/inspect#request-body

Reply:
@efjacobson you could have an overridable function for "resolveScDomain(Message $article)" or something.  less config/flags is nice if it's resolvable with code.

Reply:
@gdbrown how about a `resolveSiteUrl(Message $node)` ?

```php
$request->setSiteUrl($this->resolveSiteUrl($node));
```

Reply:
@efjacobson sure, maybe "resolveGoogleSiteUrl" since this class is handling other providers maybe.

Reply:
Great, so that will probably look like

```
    public function resolveGoogleSiteUrl(Message $node): string {
        $url = UriTemplateService::expand(
            "{$node::schema()->getQName()}.canonical", $node->getUriTemplateVars()
        );

        return parse_url($url, PHP_URL_HOST);
    }
    ```
    
    I see are using `getCanonicalUrl` in a few places. At this point should we move this method to be a shared utility in the app or just rewtite the code for this class?